src/stash-list/tab.vue

```diff
@@ -49,6 +49,15 @@ export default defineComponent({
         targetWindow(): number | undefined {
             return this.model().tabs.targetWindow.value;
         },
+        favIcon(): string {
+            if (this.tab.$selected) {
+                return '';
+            } else if (this.tab.favIconUrl) {
+                return this.tab.favIconUrl;
+            }
+            return this.model().favicons
+                .getIfExists(this.tab.url)?.value?.favIconUrl ?? '';
+        },
```

Comment:
Suggest making this return a `FaviconEntry` instead of just a `string`, and synthesizing a `FaviconEntry` if you don't have one returned by the cache.

Since `favIcon` is a computed property, Vue memoizes the return value, and re-computes the property only if any of the data dependencies have changed (`this.tab.$selected`, `.favIconUrl`, `.url`, etc.). So if we are loading a favicon from the cache (lines 58-59), and the favicon changes but the URL doesn't, you'll have to do another `.getIfExists()` lookup on the URL.

However, if you just return the FaviconEntry without peeking inside it, the `favIcon` property itself doesn't need to be re-computed and the redundant `.getIfExists()` call can be avoided. The memoized FaviconEntry--with its updated `.value` property--will be used directly.